core/src/ids.rs

```diff
@@ -224,41 +225,65 @@ impl ReservationId {
     }
 }
 
+/// Trait representing the extra info needed for program id generation.
+/// This is usually a MessageId, but it can be a handle in the form of a u32 or u64.
+pub trait ProgramNonce {
+    fn as_nonce(&self) -> &[u8];
+}
+
+impl ProgramNonce for MessageId {
+    fn as_nonce(&self) -> &[u8] {
+        self.as_ref()
+    }
+}
+
+impl ProgramNonce for u32 {
+    fn as_nonce(&self) -> &[u8] {
+        unsafe { slice::from_raw_parts(self as *const u32 as *const u8, 4) }
+    }
+}
+
+impl ProgramNonce for u64 {
+    fn as_nonce(&self) -> &[u8] {
+        unsafe { slice::from_raw_parts(self as *const u64 as *const u8, 8) }
```

Comment:
please no

not only this is platform dependant, it is also unsafe

see
https://doc.rust-lang.org/nightly/std/primitive.u64.html#method.to_le_bytes

Reply:
That was the initial thing but I couldn't return a reference so I had to allocate a Vec, which wasnt nice, though I could revert it back to that.

Reply:
You can't have platform-independent slice in memory for the integer